datafusion/expr/src/expr.rs

```diff
@@ -324,6 +319,20 @@ impl Like {
     }
 }
 
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct GetIndexedField {
```

Comment:
I think we can just re-use the docs from the `Expr::GetIndexField` enum variant here:

```suggestion
/// Returns the field of a [`arrow::array::ListArray`] or [`arrow::array::StructArray`] by key
pub struct GetIndexedField {
```